Code/src/main/java/nl/utwente/group10/haskell/type/Type.java

```diff
@@ -46,4 +70,93 @@ public final Type getFresh() {
     
     @Override
     public abstract String toString();
+    
+    /**
+     * The list type constructor.
+     */
+    public final static TypeCon ListCon = new ListTypeCon();
+    
+    /**
+     * @return a new type variable constructed from a name and optionally a number of class constraint 
+     * @param name Identifier for this type variable.
+     * @param constraints The constraints for this type
+     */
+    public final static TypeVar var(final String name, TypeClass... constraints) {
+        return new TypeVar(name, constraints);
+    }
+    
+    /**
+     * @return a new type variable conctructed from a name and a class constraint set. 
+     * @param name Identifier for this type variable.
+     * @param constraints The set of constraints for this type
+     */
+    public final static TypeVar var(final String name, final Set<TypeClass> constraints) {
+        return new TypeVar(name, 0, constraints, null);
+    }
+
+    /**
+     * @return a new type constructor
+     * @param The name of type constructor.
+     */
+    public final static TypeCon con(String name) {
+        return new TypeCon(name);
```

Comment:
Is this method required? IIRC the 'T...' vararg syntax used below means 'one or more T'.